WebKit Swift cmake build.

adetaylor opened the pull request:
#### bd13058f882a1c9516182b8028f24780d47eff60
<pre>
WebKit Swift cmake build.
<a href="https://bugs.webkit.org/show_bug.cgi?id=298090">https://bugs.webkit.org/show_bug.cgi?id=298090</a>
<a href="https://rdar.apple.com/159435279">rdar://159435279</a>

Reviewed by Don Olmstead.

This includes cmake rules to build a Swift toolchain test feature in WebKit.
To build that feature, pass --swift-demo-uri-scheme to build-webkit.

Canonical link: <a href="https://commits.webkit.org/299854@main">https://commits.webkit.org/299854@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd527da7542e29a7687aa3ded25eba170a3d7e22

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120435 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30781 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126810 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72510 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2c2e616f-5e30-4889-bcfb-6faca9c1deb0) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40826 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48706 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91464 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60751 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/086fb0d9-78e5-41ab-8f86-b0fcaec25d45) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123387 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32600 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107956 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72015 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/efd11c01-6242-4df7-a04f-847fd781cbbf) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31637 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70424 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/112561 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102081 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26244 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129691 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/118951 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47356 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35934 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100083 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47722 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104138 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99925 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25374 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45370 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23403 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44002 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47218 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52923 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/147650 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46686 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37928 "Found 1 new JSC binary failure: testapi, Found 18645 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default, ChakraCore.yaml/ChakraCore/test/Function/StackArgsWithFormals.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50033 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48373 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

dpogue commented on Source/WebKit/PlatformGTK.cmake:
```diff
@@ -10,6 +13,10 @@ if (USE_SKIA)
 
+if (NOT DEFINED ENV{SWIFTC})
+    message(FATAL_ERROR "Must define SWIFTC environment variable to point to swiftc")
+endif ()
```
Is this check needed?

My understanding is that when you declare Swift as a language in the `project(WebKit)` line, CMake will try to automatically find the Swift compiler (and use the `SWIFTC` environment variable as a hint) and fail the configuration step if it cannot.

adetaylor replied:
Agreed, not needed, will remove

mcatanzaro reviewed (commented):
Looks reasonable to me. I'll try this myself soon.

Ideally we would try to move as much of this boilerplate as possible into WebKitMacros.cmake.

donny-dont commented on Source/cmake/WebKitFeatures.cmake:
```diff
@@ -439,6 +447,10 @@ macro(CREATE_CONFIGURATION_HEADER)
     file(REMOVE "${CMAKE_BINARY_DIR}/cmakeconfig.h.tmp")
 endmacro()
 
+macro(GET_WEBKIT_COMPILE_DEFINITIONS _var_name)
```
This is the place for all compiler flags in the CMake. https://github.com/WebKit/WebKit/blob/main/Source/cmake/WebKitCompilerFlags.cmake please see if what you want is there or if it needs to be added put it there and use the naming convention in it.

adetaylor replied:
I think this was just badly named. The new macro returns config options rather than all the compiler flags, so I think this is the right place for it. I've renamed the new macro.

donny-dont reviewed (approved):
I don't have a swift compiler so maybe @mcatanzaro can see if this compiles or not but I don't see anything wrong with the CMake bits other than my nit pick.

So if @mcatanzaro is happy this can land after feedback is resolved.

adetaylor committed:
WebKit Swift cmake build.
https://bugs.webkit.org/show_bug.cgi?id=298090
rdar://159435279

Reviewed by Don Olmstead.

This includes cmake rules to build a Swift toolchain test feature in WebKit.
To build that feature, pass --swift-demo-uri-scheme to build-webkit.

Canonical link: https://commits.webkit.org/299854@main
